src/lis3dh-motion-detection.cpp

```diff
@@ -191,7 +191,64 @@ status_t LIS3DH::writeRegister(uint8_t offset, uint8_t dataToWrite) {
     returnError = IMU_HW_ERROR;
   }
 
+	_DEBBUG("Write register 0x", offset, " = ", dataToWrite);
 	return returnError;
+
+	delay(100);             // some users reported hangs without delay.
```

Comment:
Can you please refer back to the users who reported hangs? I would avoid delay at any cost.

Reply:
See last message https://community.st.com/s/question/0D50X00009XkgOnSAJ/lis3dh-inertial-interruptwakeup-not-detected

Also in documentation, in self test they rote that they wait approx 90ms for reading to calm down.